packages/vue-pdf-viewer/src/components/PDFViewer.vue

```diff
@@ -161,6 +161,24 @@ export default {
                 }
             }
         },
+        async getPdfData() {
```

Comment:
What's the plan how these functions are accessed?

Reply:
`getPdfData` is used to get the actual pdf data seen in the viewer (with filled form) and then upload and/or sign it. `formValuesHaveChanged` is used to check if we even need to get the data from the viewer or we still can use the DA pdf. See [https://github.com/certifaction/certifaction-web/pull/1296/files#r1603132670](here). There might be some changes because of improvements and cleanup but I need this PR merged to get it to on-demand.

Reply:
Ah ok, so it's through the component ref. I wonder if there is a better option, using function directly through a ref always feels a bit hacky 😅

Reply:
Yeah I tried to avoid it where I could. For example for the processing where I don't have the viewer available I just use the pdflib separately and load the document again. But for places where the viewer is available and the document already loaded it felt wrong to load it again when we already have it. But there is definitely room for improvements and I'll check again when cleaning up and improving more.